CBinding/CLang/CLangManager.cs

```diff
@@ -0,0 +1,528 @@
+				if (translationUnits.ContainsKey (fileName)) {
+					return translationUnits [fileName];
+				} else {
+					AddToTranslationUnits (proj, fileName, unsavedFiles);
+					return translationUnits [fileName];
+				}
+			}
+		}
+
+		/// <summary>
+		/// Does the "real" translation unit creating, adds it to TranslationUnits collection, from which its later available.
+		/// </summary>
+		/// <param name="proj">
+		/// A <see cref="CProject"/> reference: the project which the TU is associated. Contains the compiler arguments, in its configurations
+		/// </param>
+		/// <param name="fileName">
+		/// A <see cref="string"/>: The filename associated with the translation unit. Basically the source file's name
+		/// </param>
+		/// <param name = "unsavedFiles">
+		/// A <see cref="CXUnsavedFile"/> array: array with the contents of unsaved files in IDE. Safe to be a null sized array - CDocumentParser.Parse reparses the TU with properly initialized unsaved files.
+		/// </param>
+		void AddToTranslationUnits (CProject proj, string fileName, CXUnsavedFile[] unsavedFiles)
+		{
+			lock (SyncRoot) {
+				ClangCCompiler compiler = new ClangCCompiler ();
+				CProjectConfiguration active_configuration =
+					(CProjectConfiguration)proj.GetConfiguration (IdeApp.Workspace.ActiveConfiguration);
+				while (active_configuration == null) {
+					Thread.Sleep (20);
+					active_configuration = (CProjectConfiguration)proj.GetConfiguration (IdeApp.Workspace.ActiveConfiguration);
+				}
+				string[] args = compiler.GetCompilerFlagsAsArray (proj, active_configuration);
+				try {
+					translationUnits.Add (fileName, clang.createTranslationUnitFromSourceFile (
+						index,
+						fileName,
+						args.Length,
+						args,
+						Convert.ToUInt32 (unsavedFiles.Length),
+						unsavedFiles)
+					);
+					UpdateDatabase (proj, fileName, translationUnits[fileName]);
+				} catch (ArgumentException) {
+					Console.WriteLine (fileName + " is already added, not adding");
+				}
+			}
+		}
+
+		/// <summary>
+		/// Updates Symbol database associated with the fileName
+		/// </summary>
+		/// <param name="proj">
+		/// A <see cref="CProject"/> reference: the project which the symbol database is associated.
+		/// </param>
+		/// <param name="fileName">
+		/// A <see cref="string"/>: The filename associated with the symbol database. Basically the source file's name
+		/// </param>
+		/// <param name = "TU">
+		/// A <see cref="CXTranslationUnit"/>: the translation unit which's parsed content fills the symbol database
+		/// </param>
+		/// <param name = "cancellationToken"></param>
+		public void UpdateDatabase(CProject proj, string fileName, CXTranslationUnit TU, CancellationToken cancellationToken = default(CancellationToken))
+		{
+			lock (SyncRoot) {
+				proj.db.Reset (fileName);
+				CXCursor TUcursor = clang.getTranslationUnitCursor (TU);
+				TranslationUnitParser parser = new TranslationUnitParser (proj.db, fileName, cancellationToken, TUcursor);
```

Comment:
Clarification - the type is obvious in cases like `var f = new Foo()`. I wouldn't use var in cases like `Foo f = DoStuff()`.